### PR TITLE
 use Secret to pull an image from a private container image registry

### PR DESCRIPTION
**Pull request by waakanni**

### CODE REVIEW

File: charts/ghga-common/templates/_image-pull-secrets.tpl [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{{- define "ghga-common.image-pull-secrets" -}}
+{{- $secretName := .Values.imagePullSecretName -}}
+{{- if $secretName -}}
```
Comment (ckaipf) [inline charts/ghga-common/templates/_image-pull-secrets.tpl:3]: I think it is not necessary to bind it to a variable here.

**Comment by ckaipf** [comment]
Why did you close the old PR, it was almost finished, or?

File: charts/ghga-common/templates/_image-pull-secrets.tpl [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{{- define "ghga-common.image-pull-secrets" -}}
+{{- if .Values.imagePullSecretName -}}
+imagePullSecrets:
+  - name: {{ .Values.imagePullSecretName }}
```
Comment (ckaipf) [inline charts/ghga-common/templates/_image-pull-secrets.tpl:4]: Please make this more generic, like this we can only add one secret (which is fine for the moment). However, the requirements may change in the future. Check out [this example](https://stackoverflow.com/questions/62901288/how-to-use-imagepullsecrets-in-helm-3).

**Comment by waakanni** [comment]
starting a fresh again

**Pull request closed without merging**